cypress/integration/CTS/SearchResultsPageTests/index.js

```diff
@@ -0,0 +1,189 @@
+});
+
+And('{string} button has a link {string}', (chatNowButton, link) => {
+    cy.get('#proactive-chat-form').should('have.attr', 'action', link);
+    cy.get('#chat-button').should('have.text', chatNowButton);
+});
+
+
+When('user clicks on Chat Now button', () => {
+    cy.get('form').then((form$) => {
+        form$.on('submit', (e) => {
+            e.preventDefault();
+            cy.get('button#chat-button').click();
+        });
+    });
+});
+
+Then('the chat is opened in a new tab', () => {
+    cy.get('#proactive-chat-form').should('have.attr', "target", "_blank");
+});
+
+And(`all result item's checkboxes are not checked`, () => {
+    cy.get('.results-list .cts-checkbox .cts-checkbox__label').each(($el) => {
+        const win = $el[0].ownerDocument.defaultView;
+        // use getComputedStyle to read the pseudo selector
+        const before = win.getComputedStyle($el[0], ':before');
+        // read the value of the `content` CSS property
+        const contentValue = before['background-image'];
+        // the returned value will have double quotes around it, but this is correct
+        expect(contentValue).to.eq('none');
+    });
+});
+
+And('{string} checkbox is displayed', (selectAllCheckbox) => {
+    cy.get(`div.cts-checkbox.check-all label[for='select-all-cbx-top']`).should('be.visible').and('have.text', selectAllCheckbox);
+});
+
+When('user selects {string} checkbox', (selectAllCheckbox) => {
+    cy.get(`div.cts-checkbox.check-all label[for='select-all-cbx-top']`).contains(selectAllCheckbox).click();
+});
+
+Then(`all result item's checkboxes are checked`, () => {
+    cy.get('.results-list .cts-checkbox .cts-checkbox__label').each(($el) => {
+        const win = $el[0].ownerDocument.defaultView;
+        // use getComputedStyle to read the pseudo selector
+        const before = win.getComputedStyle($el[0], '::before');
+        // read the value of the `content` CSS property
+        const contentValue = before['background-image'];
+        // the returned value will have double quotes around it, but this is correct
+        expect(contentValue).not.to.eq('none');
+    });
+});
+
+When(`user selects {int} result's checkbox`, (num) => {
+    cy.get(`div.cts-checkbox.check-all label`).eq(num - 1).click();
+});
+
+Then('{string} checkbox is not checked', (selectAllCheckbox) => {
+    cy.get('label').contains(selectAllCheckbox).parent().find('input').should('not.be.checked');
+});
+
+And(`{int} result's checkbox is not checked`, (num) => {
+    cy.get('label').eq(num - 1).parent().find('input').should('not.be.checked');
+});
+
+Then('{string} checkbox is checked', (selectAllCheckbox) => {
+    cy.get('label').contains(selectAllCheckbox).parent().find('input').should('be.checked');
+});
+
+And(`{int} result's checkbox is checked`, (num) => {
+    And(`{int} result's checkbox is not checked`, (num) => {
+        cy.get('label').eq(num - 1).parent().find('input').should('be.checked');
```

Comment:
same - change locator

Reply:
done

Reply:
this locator is still pointing to the 'all' checkbox 
<img width="1084" alt="Screen Shot 2022-01-10 at 10 58 49 AM" src="https://user-images.githubusercontent.com/67657582/148797207-3ee0cb24-7d1b-4a45-b8bc-9667c940f136.png">